backend/events/views.py

```diff
@@ -1,3 +1,35 @@
-from django.shortcuts import render
+from rest_framework import generics, status
+from rest_framework.exceptions import ValidationError
+from rest_framework.mixins import CreateModelMixin, ListModelMixin
+from rest_framework.response import Response
+
+from .models import Event
+from .serializers import EventRegistrationSerializer, EventSerializer
+
 
 # Create your views here.
+class EventAPIView(generics.GenericAPIView, ListModelMixin, CreateModelMixin):
+    queryset = Event.objects.all()
+
+    def get_serializer_class(self):
+        if self.request.method == "POST":
+            return EventRegistrationSerializer
+        return EventSerializer
+
+    def get(self, request, *args, **kwargs):
+        try:
+            return self.list(request, *args, **kwargs)
+        except Exception as e:
+            return Response(
+                {"error": str(e)}, status=status.HTTP_500_INTERNAL_SERVER_ERROR
+            )
```

Comment:
```suggestion
    def get(self, request, *args, **kwargs):
        try:
            if 'pk' in kwargs:  # If pk is provided, retrieve a single instance
                return self.retrieve(request, *args, **kwargs)
            return self.list(request, *args, **kwargs)  # Otherwise, list all instances
        except Exception as e:
            return Response(
                {"error": str(e)}, status=status.HTTP_500_INTERNAL_SERVER_ERROR
            )
```